### PR TITLE
Fixed makefile to use pkg-config instead of ncurses5-config

### LINKED ISSUE #22329: Github pull of cataclysm master fails to link.
Game version: 
commit 7e56bbe470b4243543444c4b6ba6020910ec51c6 (HEAD -> master, upstream/master)
Merge: 67bdf6e5a0 1acb107c97
Author: Angela Graves <rivet.the.zombie@gmail.com>
Date:   Thu Nov 2 00:18:59 2017 -0500

    Merge pull request #22324 from cainiaowu/lye-recipe-1
    
    fix missing lye recipe

### Operating system:
LInux MINT x64 
Tiles or curses:
Curses 
GCC: 7.2.0
ld: (GNU Binutils) 2.29.1

### Mods active:
N/A 
### Expected behavior
Finish compile. 
### Actual behavior
j/iuse_software_kitten.o obj/char_validity_check.o obj/consumption.o obj/worldfactory.o obj/speech.o obj/creature.o
/usr/bin/ld: obj/input.o: undefined reference to symbol 'wtimeout'
/usr/lib/libtinfo.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make: *** [Makefile:686: cataclysm] Error 1
make: *** Waiting for unfinished jobs....

### Steps to reproduce the behavior
Compile game with RELEASE=1

### PR DESCRIPTION
**Pull request by Treah**
This fixes #22329 
Changed makefile to use the more standard pkg-config instead of the ncurses specific one.  The make file should probably be changed a bit more so there is not 2 checks for this program which I believe there are but this was just a quick edit to get newer versions of ncurses to build the game.

### CODE REVIEW

**Comment by kevingranade** [comment]
We need some validation that this doesnt break our builds on older systems.

**Comment by Treah** [comment]
@kevingranade 
It should not since there is a fall though if statement there that will set the build flags back to the default -lncurses if the pkg-config is not found. However pkg-config should work just fine on older versions of ncurses.  If you want me to test this on an older system let me know how far back you would like to go and ill setup a VM with something running. I know ive tried to build the game on something from 2010 before and it didn't work even before this change.

**Comment by kevingranade** [comment]
Any particular reason you didn't leave the invocation of ncursesw5-config in as a fallback?
If that's there it's a no-brainer to merge since systems without pkg-config can still use the old behavior.

> I know ive tried to build the game on something from 2010 before and it didn't work even before this change.

I'm incredibly sceptical this is related to the current issue since ncurses6 wasn't released until 2015 :/

**Comment by Treah** [comment]
@kevingranade 

> Any particular reason you didn't leave the invocation of ncursesw5-config in as a fallback?
If that's there it's a no-brainer to merge since systems without pkg-config can still use the old behavior.

Because having a strict use of a package dependent configurator is pointless really, especially one that is a single version of the package. This is only for non windows hosts that are not building CROSS and from what I know pkg-config is pretty much standard across the board. Also i think the only thing it ever pops out for you in ncurses5 is -lncurses so the fall though for systems without pkg-config will default to useing the same thing that ncurses5-config outputs.

**Comment by kevingranade** [comment]
> Because having a strict use of a package dependent configurator is pointless really

You are *guessing*(!!!) that pkg-config is available everywhere, it is not.  You are going out of your way to sabotage someone's build based solely on your idea of "the right way to do it".  If your assumption about pkg-config being universally available is incorrect, it is definitely not pointless.

Additionally, I didn't say to "strictly" use the package dependent configurator, if I wanted that behaviour I'd just close the PR.  I pointed out my issue would be resolved if you kept its invocation as a fallback.

> i think the only thing it ever pops out for you in ncurses5 is -lncurses

"i think" is not a sufficient justification.  You know this is configurable by the distribution's packaging of ncurses right? There is very nearly no way to validate that this is the case on all systems where ncursesw5-config is present.

> its been inside linux systems pretty much since inception

This is flat-out wrong. Linux has a much longer history than pkg-config, and even post-2000 it has not had universal adoption.  Also we build just fine on BSDs and other semi-POSIX systems, so even if it were universal on linux systems, that's not a sufficient reason to remove it.

> I guess you could add in the package specific one but I think if we do that it would be better to make it agnostic and not dependent on a particular version.

Go ahead, but if you can't or aren't interested in doing it, don't use it as a justification to remove it.  The status quo is that we invoke the ncurses configurator and it works on many, many systems.  Don't break that.

**Comment by Treah** [comment]
@kevingranade 
I apologize if you are under the impression im trying to push some kind of agenda, I am not.  I know that this system (pkg-config) was designed to prevent having to put package specific types of configurators into a make file. Pkg-config is also used in another part of the make file for the sdl builds along with the sdl config. 

I agree with you that "i think" is not good enough justification for anything but again I cannot test everything out there and this is why this PR is listed with a CR tag. I am open to feedback on it as its changing the make file which is very important to the build process. 

I can go ahead and update the makefile with changes to allow the package specific one to be picked up in the event that pkg-config is not on the system, there is a place in the SDL part of the make that does that as well so having it default down is a good idea in the event the pkg-config fails or is not present. I will see if a wild card or something can be used to get the right binary to run. Do we still want to fall though tho to just using a -lncurses if either of them fails tho? I am uncertain if there is a possibility of having ncurses on the system that does not include the library specific config binary? 

Ohh as a FYI on BSD the game does not compile on net-bsd but does on freebsd. Its possible the build is broken on all bsd systems other then that one. The error it gets on net-bsd however I no longer have, but if I do try and build it again there I can open a seperate issue on it.

**Comment by kevingranade** [comment]
> I am open to feedback on it as its changing the make file which is very important to the build process.

> leave the invocation of ncursesw5-config in as a fallback

**Comment by kevingranade** [comment]
What part of, "leave the ncursesw5-config behavior the same" was unclear?
You stopped passing it arguments, and you only use it to set LDFLAGS.

**Comment by Treah** [comment]
@kevingranade 
Sorry intended to do so as requested but I must have been tired and never pushed the changes. It should be good now.

**Comment by narc0tiq** [comment]
As I touched upon in #22435, `ncurses` and `ncursesw` (note the W at the end) are not the same thing -- in particular, the base `ncurses` does not have wide-character support, while the `ncursesw` does, according to [the ncurses faq](http://invisible-island.net/ncurses/ncurses.faq.html). The FAQ also mentions some systems are known to rename `ncursesw` to `ncurses` by default (I think `osxcross` does this, or more likely it's built `ncurses` from source with `--enable-widec` -- or maybe it doesn't have ncurses at all, and this is an artifact of @patternoia's setup).

I'm not entirely certain what this impacts; I know for certain we use *non-ASCII* characters (>= 128), but multi-byte characters I don't know about. Certainly, in a Unicode environment, even the line drawing characters are multi-byte (starting at code point U+2500), so I can only assume that for a UTF-8 locale `ncursesw` would be *required*, but I've never gone much deeper than what I've just mentioned: that is, 
- that `ncurses` and `ncursesw` are not the same thing, and 
- that you probably want to prefer `ncursesw` and fall back to `ncurses` if not available.

File: Makefile [hunk 1]
```diff
@@ -552,18 +552,25 @@ ifdef TILES
...
-    CXXFLAGS += $(shell ncursesw5-config --cflags)
-    LDFLAGS += $(shell ncursesw5-config --libs)
+  ifeq ($(HAVE_PKGCONFIG),1)
+    CXXFLAGS += $(shell pkg-config --cflags ncurses)
+    LDFLAGS += $(shell pkg-config --libs ncurses)
   else
     ifneq ($(TARGETSYSTEM),WINDOWS)
-      LDFLAGS += -lncurses
+  # This is here for backward comaptablity as a fallback. 
+      ifneq ($(shell which ncurses5-config 2>/dev/null),)
```
Comment (narc0tiq) [inline Makefile]: This fallback doesn't have the same semantics as the original code that was removed, which was inside a further `ifeq ($(CROSS),)` (i.e., "if `$(CROSS)` is unset").

I rather imagine you should've kept the `HAVE_NCURSESW5CONFIG` variable and kept the check in the same place it was before -- only prefixing the check for `pkg-config`. I think this is actually what caused the error in #22435.

Comment (Treah) [reply Makefile]: Yes it does. The code for doing anything with ncurses only runs if your not doing cross. All that does there is check to see if cross is being used and if not set the flag. This also would not have caused the error in that as this was only merged in 17 hours ago as of my posting.

Comment (narc0tiq) [reply Makefile]: 17 hours ago, a push to master by kevingranade contained this PR, as well as #22423, which fixed the previous error with the OSX build, reported at #22421.

If you follow the logic in the Makefile, the blocks are (in some kind of pseudocode):
```
if TILES is set {
    // stuff that wasn't changed
} else {
    if CROSS is unset {
        if $(shell which pkg-config) is not empty {
            HAVE_PKGCONFIG = 1
       }
    }

    if HAVE_PKGCONFIG == 1 {
        // ...use pkg-config...
    } else {
        if TARGETSYSTEM is not "WINDOWS" {
            if $(shell which ncurses5-config) is not empty {
                // ...use ncurses-config...
            } else {
                // add "-lncurses" to LDFLAGS
            }
        }

        if OSXCROSS is defined {
            // ... add references to $(LIBSDIR)/ncurses/{lib,include} to {LD,CXX}FLAGS
        }
    }
}
```

**Pull request merged**